plugins/auth-backend/src/providers/oidc/provider.ts

```diff
@@ -56,18 +56,21 @@ type OidcImpl = {
   client: Client;
 };
 
-type AuthResult = {
+export type OidcAuthResult = {
   tokenset: TokenSet;
   userinfo: UserinfoResponse;
 };
 
+export type OidcSignInResolver = SignInResolver<OidcAuthResult>;
```

Comment:
Feeling that it doesn't really make sense to export these little alias types so much. It's got very little value to committing to this as public api from this package.

Reply:
These types are missing if I want to define my functions like:

AuthResult and AuthHandlerResult is missing.

```typrscript
private authHandler: (input: AuthResult) => Promise<AuthHandlerResult>;
```

Of course I can use create my functions directly while creating a function. If you think this should be way to go, I can clean-up and export only necessary types like OidcProviderOptions
```typescript
    createOidcProvider({
      authHandler: async ({ userinfo, ctx }) => {   do something.. }
      signIn: {
        resolver: async ({ result }) => {   do something.. }
      },
    });
```

Reply:
One more option would be to export SignInResolver and AuthHandler instead of these tiny types, but it is a general export and is not only related to OIDC provider :)

Reply:
So I think it's fine to export the `OidcAuthResult` but not sure that we should export the `OidcSignInResult` or `OidcAuthHandler` as you can just write those types out yourself. They are too tiny to provide any value I think.

The `SignInResolver` type and `AuthHandler` types are already exported from this package already, so you could just use things like `SignInResolver<OidcAuthResult>` in your own code.

They're very specific to OIDC, and they're only used in this file and anyone who wants to write additional handlers. Thinking that it's not worth it to export them and create another type from this package that might have to deprecate or change in the future.

Or am I totally on the wrong track here an missed something?

Reply:
I agree, the only missing part is here: **plugins/auth-backend/src/providers/index.ts**  SignInResolver and AuthHandler is not exported so I can not import on my own code. Instead of exporting this tiny types, I think it makes sense to export these base type so the caller can define the types itself. If it is ok for you, I will revert this part and export the base types in **plugins/auth-backend/src/providers/index.ts**

Reply:
ah sorry  - i was under the impression that those were already exported. Yep please do. /cc @Rugvip think we missed this.